tools/gen-providers.ts

```diff
@@ -499,6 +500,27 @@ const sse_rule: RecordDecl = {
   ],
 };
 
+const and: RecordDecl = {
```

Comment:
Those are usually prefixed with enclosed type. Something like `aws_s3_bucket_lifecycle_configuration_filter_and`. Same for `filter`.

Reply:
And `generateParams` for it.

Reply:
Done